docs/whats-new/dotnet-aspire-9.4.md

```diff
@@ -0,0 +1,1919 @@
+
+### 📦 Azure Storage blob container creation changes
+
+Azure Storage blob container creation has been moved from specialized blob storage resources to the main storage resource for better consistency:
+
+```csharp
+// ❌ Before (obsolete):
+var storage = builder.AddAzureStorage("storage");
+var blobs = storage.AddBlobs("blobs");
+var container = blobs.AddBlobContainer("images");     // Obsolete
+
+// ✅ After (recommended):
+var storage = builder.AddAzureStorage("storage");
+var container = storage.AddBlobContainer("images");   // Direct on storage resource
+```
+
+**Migration impact**: Use `AddBlobContainer()` directly on `AzureStorageResource` instead of on specialized blob storage resources.
+
+### 🔐 Keycloak realm import simplification
+
+The `WithRealmImport` method signature has been **simplified by removing the confusing `isReadOnly` parameter**:
+
+```csharp
+// ❌ Before (deprecated):
+var keycloak = builder.AddKeycloak("keycloak")
+    .WithRealmImport("./realm.json", isReadOnly: false);  // Confusing parameter
+
+// ✅ After (recommended):
+var keycloak = builder.AddKeycloak("keycloak")
+    .WithRealmImport("./realm.json");  // Clean, simple API
+
+// If you need explicit read-only control:
+var keycloak = builder.AddKeycloak("keycloak")
+    .WithRealmImport("./realm.json", isReadOnly: true);  // Still available as overload
+```
+
+**Migration impact**: Remove the `isReadOnly` parameter from single-parameter `WithRealmImport()` calls - the method now defaults to appropriate behavior. Use the two-parameter overload if explicit control is needed.
+
+### 🔧 Milvus configuration method updates
+
+Milvus configuration has been updated with more descriptive method names:
+
+```csharp
+// ❌ Before (deprecated):
+var milvus = builder.AddMilvus("milvus")
+    .WithConfigurationBindMount("./milvus.yaml");  // Old method name
+
+// ✅ After (recommended):
+var milvus = builder.AddMilvus("milvus")
+    .WithConfigurationFile("./milvus.yaml");  // Method renamed for clarity
+```
+
+**Migration impact**: Update method calls to use `WithConfigurationFile` instead of `WithConfigurationBindMount` for Milvus configuration.
+
+### 🔒 Azure Cosmos DB connection string property changes
+
+Azure Cosmos DB connection string handling has been updated to use the new connection string output approach:
+
+```csharp
+// ❌ Before (obsolete property):
+var cosmos = builder.AddAzureCosmosDB("cosmos");
+// cosmos.ConnectionString property is obsolete - use ConnectionStringExpression instead
+
+// ✅ After (recommended):
+var cosmos = builder.AddAzureCosmosDB("cosmos");
+var connectionString = cosmos.ConnectionStringExpression;  // Use expression-based approach
+```
+
+**Migration impact**: Replace direct `ConnectionString` property access with `ConnectionStringExpression` for Azure Cosmos DB resources.
+
```

Comment:
```suggestion
```

I don't think this Obsoletion is worth calling out. An extremely small number of customers would even be doing this. Plus also, there is no `cosmos.ConnectionString` or `cosmos.ConnectionStringExpression` properties. It needs to `.Resource` in the middle.